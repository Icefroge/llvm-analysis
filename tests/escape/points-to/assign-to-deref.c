extern int a;
extern int b;
extern int c;
int * p;
int ** pp;
int * q;
int ** pq;

void f() {
  int *** x;
  int ** y;
  int * r = &c;

  p = &a;
  q = &b;
  pp = &p;
  pq = &q;
  x = &pq;
  y = *x;
  // In this loop, *y is q.  Since q is global, the edges from it are
  // *not* killed and it can point to a, b, or c at the end of the
  // function.
  if(a > 10) {
    *y = *pp;
  }
  else {
    *y = r;
  }
}
