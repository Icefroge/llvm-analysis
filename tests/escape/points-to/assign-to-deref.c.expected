fromList [ ("@pp", fromList ["@p"])
         , ("@pq", fromList ["@q"])
         , ("@p", fromList ["@a"])
         , ("@q", fromList ["@a", "@b", "@c"])
         , ("%x", fromList ["@pq"])
         , ("%y", fromList ["@q"])
         , ("%r", fromList ["@c"])
         ]